src/frontend/src/expr/function_call.rs

```diff
@@ -135,6 +139,31 @@ impl FunctionCall {
                     .collect::<Result<Vec<_>>>()?;
                 Ok(DataType::Varchar)
             }
+
+            ExprType::SplitPart => {
+                let expected = 3;
+                let actual = inputs.len();
+                if expected != actual {
+                    return Err(ErrorCode::BindError(format!(
+                        "Function `SplitPart` takes exactly {} arguments ({} given)",
+                        expected, actual
+                    ))
+                    .into());
+                }
+
+                inputs = inputs
+                    .into_iter()
+                    .enumerate()
+                    .map(|(i, input)| match i {
+                        0 => input.cast_implicit(DataType::Varchar),
+                        1 => input.cast_implicit(DataType::Varchar),
+                        2 => input.cast_implicit(DataType::Int64),
+                        _ => todo!(),
```

Comment:
Since we had check length, what should be insert to fill to clause?

Reply:
Given this function has a fixed signature: `(varchar, varchar, i32) -> varchar`, add it in `type_inference::build_type_derive_map` rather than manually checking here.

Functions listed here are all variadic. Maybe you can do me a favor and add this as comment in this function.

Reply:
thx. [41909ae](https://github.com/singularity-data/risingwave/pull/2822/commits/41909ae9abf146401014e7d4cc8f5ed40b9333fa)